File: 백준/Silver/20300. 서강근육맨/서강근육맨.py
n = int(input())

loss = list(map(int, input().split()))
loss.sort()
rloss = sorted(loss, reverse=True)

m = 0

if n % 2 != 0:
    m = rloss[0]
    del rloss[0]

for i in range(n // 2):
    temp = loss[i] + rloss[i]
    if temp > m:
        m = temp

print(m)

File: 백준/Silver/20300. 서강근육맨/서강근육맨.swift
var n = Int(readLine()!)!
var result = -Int.max
if let input = readLine() {
    let arr = input.split(separator: " ").compactMap { Int($0) }.sorted()
    var cnt = 0
    let end = n % 2 == 1 ? n - 2 : n - 1

    for i in 0..<(n / 2) {
        let temp = arr[i] + arr[end - i]
        result = max(temp, result)
    }
    
}
print(result)